rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/util/EvaluatorUtil.java

```diff
@@ -148,17 +149,16 @@ private static String formatMessage(StaticError e) {
         return "Static error: " + e.getMessage();
     }
 
-    public static CompletableFuture<Evaluator> makeFutureEvaluator(ExecutorService exec, IBaseTextDocumentService docService, BaseWorkspaceService workspaceService, IBaseLanguageClient client, String label, PathConfig pcfg, boolean addRascalCore, final String... imports) {
+    public static CompletableFuture<Evaluator> makeFutureEvaluator(ExecutorService exec, IBaseTextDocumentService docService, BaseWorkspaceService workspaceService, IBaseLanguageClient client, String label, IRascalMonitor monitor, PathConfig pcfg, boolean addRascalCore, final String... imports) {
         return CompletableFuture.supplyAsync(() -> {
             Logger customLog = LogManager.getLogger("Evaluator: " + label);
-            IRascalMonitor monitor = new LSPIDEServices(client, docService, workspaceService, customLog);
+            IDEServices services = new LSPIDEServices(client, docService, workspaceService, customLog, monitor);
             boolean jobSuccess = false;
             try {
-                monitor.jobStart("Loading " + label);
+                services.jobStart("Loading " + label);
```

Comment:
This "Loading " job does not have steps. Better remove it; it will make more sense if we leave it to the internal evaluators "loading modules" job.

Reply:
There is a `doImport` call in this method that can now take the entire array of `imports` as a parameter. So the loop can be dropped and the internal monitor can do each module as a `jobStep`

Reply:
I'm going to add job steps, as the reason why this monitor is the top-level one is that it's part of the title of this progress bar. So I want a pretty one, and not one that says which module it's loading (you can still see that in the details.)

Reply:
Reason's I'm not doing that `doImport` call:

1. the current code allows me to print a more accurate error message of which root module went wrong
2. the `doImport` with the varargs is not in a release rascal jar yet, so we can always migrate to that later.

Reply:
Just know that every doImport call will now produce its own progress bar; or we go back to the steps without starts situation.

Reply:
Note that the exception will tell you which module went wrong anyhow.